ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts/overlay.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+
+from functools import cached_property
+from typing import TYPE_CHECKING
+
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
+
+if TYPE_CHECKING:
+    from .eos_designs_facts import EosDesignsFacts
+
+
+class OverlayMixin:
+    """
+    Mixin Class used to generate some of the EosDesignsFacts.
+    Class should only be used as Mixin to the EosDesignsFacts class
+    Using type-hint on self to get proper type-hints on attributes across all Mixins.
+    """
+
+    @cached_property
+    def evpn_role(self: EosDesignsFacts) -> str | None:
+        """
+        Exposed in avd_switch_facts
+        """
+        return self.shared_utils.evpn_role
```

Comment:
For these simple ones I think we could not write a function with a special getattr that would retrieve from shared_utils and set as cahced_property on the class a list of allowed properties to avoid having to write these 5 boilerplates lines everytime.

Happy to have a look at this if you want

Reply:
For now let's just keep this. There are small variations, and we loose type hints etc if we add them directly. I think we should wait with optimizing it until we get more of our planned changes done.